From 6f9afc6e4f7bcb32e461719502b379a7794fca2b Mon Sep 17 00:00:00 2001
From: Xiaowen Liu <xiaowen.liu@freescale.com>
Date: Thu, 15 Oct 2015 06:39:36 -0400
Subject: [PATCH 15/22] MA-7089-1 Upgrade android to Mashmallow 6.0 version.

modify init.rc according to changes in lp5.1.
1) remove ril-daemon service.
2) comment on property:ro.debuggable=1.
3) change console service property.
4) adjust virtual memory configuration.
5) create some file node.

Signed-off-by: Xiaowen Liu <xiaowen.liu@freescale.com>
---
 rootdir/init.rc | 38 +++++++++++++++++++++-----------------
 1 file changed, 21 insertions(+), 17 deletions(-)

diff --git a/rootdir/init.rc b/rootdir/init.rc
index 317207c..9dab39c 100644
--- a/rootdir/init.rc
+++ b/rootdir/init.rc
@@ -53,6 +53,7 @@ on init
     mkdir /cache 0770 system cache
     mkdir /config 0500 root root
 
+    mkdir /device 0771 system system
     # Mount staging areas for devices managed by vold
     # See storage config details at http://source.android.com/tech/storage/
     mkdir /mnt 0755 root system
@@ -63,13 +64,17 @@ on init
     mkdir /mnt/secure/asec 0700 root root
     mkdir /mnt/asec 0755 root system
     mkdir /mnt/obb 0755 root system
-    mkdir /mnt/media_rw 0750 root media_rw
+    mkdir /mnt/media_rw 0751 root media_rw
     mkdir /mnt/user 0755 root root
     mkdir /mnt/user/0 0755 root root
     mkdir /mnt/expand 0771 system system
 
     # Storage views to support runtime permissions
     mkdir /storage 0755 root root
+    mkdir /mnt/media_rw/extsd 0755 system system
+    symlink /mnt/media_rw/extsd /storage/extsd
+    mkdir /mnt/media_rw/udisk 0755 system system
+    symlink /mnt/media_rw/udisk /storage/udisk
     mkdir /mnt/runtime 0700 root root
     mkdir /mnt/runtime/default 0755 root root
     mkdir /mnt/runtime/default/self 0755 root root
@@ -82,6 +87,10 @@ on init
     symlink /storage/self/primary /sdcard
     symlink /mnt/user/0/primary /mnt/runtime/default/self/primary
 
+    # shared memory used by VPU to distinguish VPU instance.
+    mkdir /mnt/shm  0775 media syste
+    mount tmpfs tmpfs /mnt/shm mode=0775,uid=1013,gid=1000,size=1m
+
     # memory control cgroup
     mkdir /dev/memcg 0700 root system
     mount cgroup none /dev/memcg memory
@@ -259,6 +268,8 @@ on post-fs
     # Create /cache/recovery in case it's not there. It'll also fix the odd
     # permissions if created by the recovery system.
     mkdir /cache/recovery 0770 system cache
+    # This may have been created by the recovery system with the wrong context.
+    restorecon /cache/recovery
 
     #change permissions on vmallocinfo so we can grab it from bugreports
     chown root log /proc/vmallocinfo
@@ -303,6 +314,7 @@ on post-fs-data
 
     # Avoid predictable entropy pool. Carry over entropy from previous boot.
     copy /data/system/entropy.dat /dev/urandom
+    mkdir /data/media 0770 media_rw media_rw
 
     # create basic filesystem structure
     mkdir /data/misc 01771 system misc
@@ -343,6 +355,7 @@ on post-fs-data
     mkdir /data/app-asec 0700 root root
     mkdir /data/app-lib 0771 system system
     mkdir /data/app 0771 system system
+    mkdir /data/app/ing 0771 root root
     mkdir /data/property 0700 root root
     mkdir /data/tombstones 0771 system system
 
@@ -398,7 +411,7 @@ on post-fs-data
     # must uncomment this line, otherwise encrypted filesystems
     # won't work.
     # Set indication (checked by vold) that we have finished this action
-    #setprop vold.post_fs_data_done 1
+    setprop vold.post_fs_data_done 1
 
 on boot
     # basic network init
@@ -420,7 +433,8 @@ on boot
     chmod 0664 /sys/module/lowmemorykiller/parameters/minfree
 
     # Tweak background writeout
-    write /proc/sys/vm/dirty_expire_centisecs 200
+    write /proc/sys/vm/dirty_expire_centisecs 50
+    write /proc/sys/vm/dirty_writeback_centisecs 100
     write /proc/sys/vm/dirty_background_ratio  5
 
     # Permissions for System Server and daemons.
@@ -574,13 +588,11 @@ service healthd /sbin/healthd
 service console /system/bin/sh
     class core
     console
-    disabled
-    user shell
-    group shell log
-    seclabel u:r:shell:s0
+    user root
+    group root
 
-on property:ro.debuggable=1
-    start console
+#on property:ro.debuggable=1
+#    start console
 
 # adbd is controlled via property triggers in init.<platform>.usb.rc
 service adbd /sbin/adbd --root_seclabel=u:r:su:s0
@@ -633,14 +645,6 @@ service debuggerd64 /system/bin/debuggerd64
     class main
     writepid /dev/cpuset/system-background/tasks
 
-service ril-daemon /system/bin/rild
-    class main
-    socket rild stream 660 root radio
-    socket sap_uim_socket1 stream 660 bluetooth bluetooth
-    socket rild-debug stream 660 radio system
-    user root
-    group radio cache inet misc audio log
-
 service surfaceflinger /system/bin/surfaceflinger
     class core
     user system
-- 
1.8.0

